src/main/scala/encry/view/state/UtxoState.scala

```diff
@@ -103,7 +103,8 @@ class UtxoState(override val version: VersionTag,
     log.debug(s"Generating proof for ${txs.length} transactions ...")
     val rootHash: ADDigest = persistentProver.digest
     if (txs.isEmpty) Failure(new Exception("Got empty transaction sequence"))
```

Comment:
Do you mean here to make a new instance of class Exception or to throw a new Exception? If second this code is wrong. def proofsForTransactions can return Failure to val (adProof: SerializedAdProof, adDigest: ADDigest) = state.proofsForTransactions(txs).get.